build-system/tasks/helpers.js

```diff
@@ -827,31 +826,24 @@ function mkdirSync(path) {
 }
 
 /**
- * Returns the list of dependencies for a given JS entrypoint.
+ * Returns the list of dependencies for a given JS entrypoint by having esbuild
+ * generate a metafile for it. Uses the set of babel plugins that would've been
+ * used to compile the entrypoint.
+ *
  * @param {string} entryPoint
+ * @param {!Object} options
  * @return {Promise<Array<string>>}
  */
-async function getDependencies(entryPoint) {
-  const stripImportAssertionPlugin = {
-    name: 'stripImportAssertion',
-    setup(build) {
-      build.onLoad({filter: /\.[cm]?js$/, namespace: ''}, async (file) => {
-        const filename = file.path;
-        const contents = (await batchedRead(filename)).contents.toString();
-        const stripped = contents.replace(/assert {type: 'json'}/g, '');
-        return {contents: stripped};
-      });
-    },
-  };
-
+async function getDependencies(entryPoint, options) {
+  const caller = options.minify ? 'minified' : 'unminified';
+  const babelPlugin = getEsbuildBabelPlugin(caller, /* enableCache */ true);
```

Comment:
Jake purposefully didn't use Babel here, because it's significantly slower. I think all that was necessary was to use the JSX esbuild loader, right?

Reply:
Here's my rationale behind using Babel in this PR:
- `getDependencies()` is called only while initially watching a main entry point, which are significantly fewer in number than JS files in total
- It doesn't really matter if the call to Babel is slightly slower because `getDependencies()` is only called once during startup
- The fact that caching is enabled for `getEsbuildBabelPlugin()` means that the subsequent transforms of the entry point will be instant (so this shouldn't actually slow down anything)
- This PR will make us immune to future breakages when a new kind of file is introduced (similar to JSX)

Wanna pass this through your BS detector and tell me if there's any reason not to leave things as they are? 😃

Reply:
Sounds good to me.

Reply:
Running through my BS Detector:

> getDependencies() is called only while initially watching a main entry point, which are significantly fewer in number than JS files in total

This one is likely BS, since we actually perform a full `esbuild.build` operation here. Meaning it still transforms every file references by the entrypoint.

> getDependencies() is called only while initially watching a main entry point, which are significantly fewer in number than JS files in total

It'll be called once _per entrypoint_ (regardless of main vs. extension) during watch mode.

> The fact that caching is enabled for getEsbuildBabelPlugin() means that the subsequent transforms of the entry point will be instant (so this shouldn't actually slow down anything)

This is a fantastic point. I forgot that it becomes essentially free when cached (which we can rely on working during local dev / watch mode). Couple with the final point, I'm sold :)